packages/cli/test/unit/commands/container/release.unit.test.ts

```diff
@@ -5,514 +5,519 @@ import {expect} from 'chai'
-      .get('/v2/testapp/web/manifests/latest')
-      .reply(200, {schemaVersion: 2, config: {digest: 'image_id'}})
-    await runCommand(Cmd, [
-      '--app',
-      'testapp',
-      'web',
-    ])
-    expect(stderr.output).to.contain('Releasing images web to testapp... done')
-    expect(stdout.output, 'to be empty')
-    api.done()
-    registry.done()
+    const {message, oclif} = error as unknown as CLIError
+    expect(message).to.equal(`This command is for Docker apps only. Run ${color.cyan('git push heroku main')} to deploy your ${color.cyan('testapp')} ${color.cyan('heroku-24')} app instead.`)
+    expect(oclif.exit).to.equal(1)
   })
 
-  it('retrieves data from a v1 schema version, no previous release', async function () {
-    const api = nock('https://api.heroku.com:443')
-      .get('/apps/testapp')
-      .reply(200, {name: 'testapp'})
-      .patch('/apps/testapp/formation', {
-        updates: [
-          {type: 'web', docker_image: 'image_id'},
-        ],
-      })
-      .reply(200, {})
-      .get('/apps/testapp/releases')
-      .reply(200, [])
-      .get('/apps/testapp/releases')
-      .reply(200, [{id: 'release_id', status: 'succeeded'}])
-    const registry = nock('https://registry.heroku.com:443')
-      .get('/v2/testapp/web/manifests/latest')
-      .reply(200, {schemaVersion: 1, history: [{v1Compatibility: '{"id":"image_id"}'}]})
-    await runCommand(Cmd, [
-      '--app',
-      'testapp',
-      'web',
-    ])
-    expect(stderr.output).to.contain('Releasing images web to testapp... done')
-    expect(stdout.output, 'to be empty')
-    api.done()
-    registry.done()
-  })
+  context('when the app is a container app', function () {
+    let registry: nock.Scope
+    beforeEach(function () {
+      api
+        .get('/apps/testapp')
+        .reply(200, {name: 'testapp', stack: {name: 'container'}})
+      registry = nock('https://registry.heroku.com:443')
+    })
 
-  it('retrieves data from a v1 schema version, with a previous release', async function () {
-    const api = nock('https://api.heroku.com:443')
-      .get('/apps/testapp')
-      .reply(200, {name: 'testapp'})
-      .patch('/apps/testapp/formation', {
-        updates: [
-          {type: 'web', docker_image: 'image_id'},
-        ],
-      })
-      .reply(200, {})
-      .get('/apps/testapp/releases')
-      .reply(200, [{id: 'old_release_id', status: 'succeeded'}])
-      .get('/apps/testapp/releases')
-      .reply(200, [{id: 'release_id', status: 'succeeded'}])
-    const registry = nock('https://registry.heroku.com:443')
-      .get('/v2/testapp/web/manifests/latest')
-      .reply(200, {schemaVersion: 1, history: [{v1Compatibility: '{"id":"image_id"}'}]})
-    await runCommand(Cmd, [
-      '--app',
-      'testapp',
-      'web',
-    ])
-    expect(stderr.output).to.contain('Releasing images web to testapp... done')
-    expect(stdout.output, 'to be empty')
-    api.done()
-    registry.done()
-  })
+    afterEach(function () {})
```

Comment:
Out of curiosity, what does this do? Why is it necessary?

Reply:
oh it isn't, nice catch!